test/addons/make-callback-domain-warning/test.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+const common = require('../../common');
+const assert = require('assert');
+const domain = require('domain');
+const binding = require(`./build/${common.buildType}/binding`);
+const makeCallback = binding.makeCallback;
+
+let latestWarning = null;
+process.on('warning', function(warning) {
+  latestWarning = warning;
+});
+
+const d = domain.create();
+
+// When domain is disabled, no warning will be emitted
+makeCallback({ domain: d }, common.mustCall(function() {
+  assert.strictEqual(latestWarning, null);
+}));
+
+d.run(common.mustCall(function() {
+  // No warning will be emitted when no domain property is applied
+  makeCallback({}, common.mustCall(function() {
+    assert.strictEqual(latestWarning, null);
+  }));
+
+  // Warning is emitted when domain property is used and domain is enabled
+  makeCallback({ domain: d }, common.mustCall(function() {
+    assert.strictEqual(latestWarning, null);
```

Comment:
I am not certain when exactly the warning should be emitted but from the comment itself it seems to me that `latestWarning` should not be `null`. If it should be, it would be good to a test that actually tests for `notStrictEqual`.

Reply:
Right, this is odd